### DISCUSSION TITLE
Fix orphaned tool qualities

### DISCUSSION THREAD

**Discussion by ghost** [cataclysmbn/Cataclysm-BN#5096] [General]
I will add onto this as additional problems are found.

The following tool qualities have no mainlined uses I can find:

Awl quality including steel and bone awls
Curved Needle
Reaming quality from Pin reamers

The following quality needs additional work:
Sewing is only used at level 1 quality but has level 4 on a tailors kit. You can use a wooden needle to stitch kevlar and leather.

The Solution? Obsolete or add recipes for them. Don't care which, I leave it up to the author. If you obsolete anything you must ensure that tailors kits no longer call for them in their recipe, as currently you require an awl and curved needle in the kit but it's still used for nothing.

Awls are used for sewing tough material such as leather. Curved needles are for neoprene. We could stand to require awls for leatherworking, curved needles for neoprene and maybe kevlar, and additionally higher sewing qualities for a lot of clothes or armors.

Reaming is for a variety of metal projects.

**Comment by KheirFerrum** [comment]
Curved needles are less about _material_ and more about application. Sewing upholstery for instance typically uses curved needles in real life because you want a tight fit and thus cannot move your hand to the other side to grab and redirect the needle.

Reams are used to get drilled holes in metal to precise diameters, this _could_ be incorporated into the game but I see very little harm in removing it given the universal and abstract nature of our crafting tools.

**Reply by ZChris13** [reply]
do the three workshop toolbox qualities actually do anything? (filing, reaming, clamping)
I couldn't find anything, although it still felt good to collect them I'm not sure why I did so or why it's still in the game

**Reply by blot-runner** [reply]
> do the three workshop toolbox qualities actually do anything? (filing, reaming, clamping)

They don't, unless it's hardcoded (which I sincerely doubt, but, well, stranger things happened). JSON search returns zero results on either these qualities or the actual items having them used anywhere in the game. Also, grinding quality is unused, but the only item having it (angle grinder) is required for several construction projects, so it's less of an issue.

Regarding the usage... Filing is too exclusive a quality to make it necessary for any common item, but if the turning is added (and there is a lathe in the game already, it just cannot be used atm), I guess they can be used together. Ditto for reamers. The tough part is creating useful recipes for that. Boring gun barrels with reamers to craft better quality firearms (is it even plausible)? Making things like gears, metal tubing and such to craft engines from scratch? Mass-producing something, e,g, a new recipe for pipes that requires a lathe, files and reamers, but takes 5 minutes per pipe, instead of one hour (do players really need to mass-produce pipes)?

As for the clamps, they are more for holding several things together (e.g. while the glue sets or to drill a hole in both of them) than for "keeping things still" as the description suggests (that's what the vise is for). Well, at least that's how I've always been using them.
So, on the one hand, their uses are many and versatile, on the other hand, they aren't exactly _indispensable_ in most scenarios, and on the third hand, to effectively use them you usually need at least two. So, I, personally, can't see how they can be saved.

**Reply by ghost** [reply]
Gun barrels are too complicated to simply bore from a pipe or block of steel, they require chemical treatment.

I got around that with exotic metal that isn't brittle.

**Reply by blot-runner** [reply]
> Gun barrels are too complicated to simply bore from a pipe or block of steel, they require chemical treatment.

That's debatable. This video shows manufacturing of precision rifle barrels from 416R stainless steel (which, according to the info I found on a random vendor site is used for "Automotive components, General engineering, Washing machine parts, Motor shafts, Gears, Valves & pumps", so it's not a far fetch to say the "lumps of steel" in CBN are eligible).
https://www.youtube.com/watch?app=desktop&v=LAf064S2Cv4
Unless I've missed something, the only chemical treatment shown is lubricating the drilled and reamed barrel before the button rifling (which, according to the video, is done with a type of soap). The whole process (cutting, reaming, machining the surface, lubricating, rifling and reheating) seems within abilities of the highly skilled survivor, all the tools except lathe are either already present or can be abstracted out.
It's not the question of whether the character CAN produce firearms of the factory-grade quality, it's the question of whether such an option can add something meaningful to the gameplay. I agree, however, that there is no real reason to tie this possible future feature with the existence of useless tools here and now. No harm in just removing reamers and filesets from the game with the possibility to reintroduce them when and if machine tools become a thing.

**Reply by ghost** [reply]
From what I was given a while back on making a rifle barrel we're looking at:

"If you're looking at using rifle ammunition with such a barrel, your barrel would become fully smooth after around 200 shots, so using water cooling in this case is near mandatory."

"Barrel is the most difficult part of all, by the way. Everything else is doable if you have a workshop stocked with machine tools."

Deep drilling (lubricating/cooling fluid is 94 % spindle oil, 4.5 % plant oil, 1.5% sulfur)
Reaming (3 passes, lubricating/cooling fluid is 89% spindle oil, 10% flax oil, 1% sulfur) and subsequent planing.
De-fatting (NaOH, 95-104 C, 15-30 minutes)
Copper plating (CuSO4 12% HCl 7% FeCl3 1% water 80%)
Mandreling
Reaming the chamber (16 operations; lubricating/cooling fluid is 100% flax oil)
Hardening in lead bath; in first - heat to 650-670 C for 2-4 minutes, in second - final heating to 830-850 C, cooling with oil under pressure of 3-4 atm, for 15 minutes.
Tempering, depending on hardness, from 480 to 600.
Cleaning from lead with sulfuric acid (10-12% concentration) 80-90 C for 15-20 minutes, wash in cold water for 1.5 - 2 minutes, neutralize in potash solution under 80-90 C, for 4-5 minutes.
(Optional) Polishing and lapping with lead
(Optional) De-fatting, electrolytically or via etching, and chrome plating.
Blackening with alkali or acid.

**Reply by ghost** [reply]
Feel free to go through the entire process of a rifle barrel and determine which steps are not critical to allowing a factory quality rifle that would fire thousands of shots without losing the rifling. I wrote it off as "Nope. I ain't doing it."

**Reply by blot-runner** [reply]
Interesting. Even if we assume the gun barrel making data I was able to find skimps of some details, it still seems the process you describe seems to be quite different from what I've been able to find. This, however, is beside the point, and, frankly, I like your process better, it both looks complex and involved and requires very few items to be added to become a thing. 


> Feel free to go through the entire process of a rifle barrel and determine which steps are not critical to allowing a factory quality rifle that would fire thousands of shots without losing the rifling.

Such exercise seems to be of little use for designing a feature for a _game_ (as opposed to firearms making simulator). If, however, I ever decide to suggest a gun-making feature for the project, I'll be sure to do my best to make the recipe: a) hard and complex enough to justify the final reward of a factory-grade firearm(s) of the player's choice; b) close enough to reality to look believable for a layman and c) relying as much as practicable on the existing tools and components. IMO it's a better approach than deciding if the spindle oil should be added to the game to make the rifling 100% true to reality, or not.
Anyway, the conversation seems to have strayed off the initial topic so far I can't even see it from where we are now. Thank you sincerely for all the info you've provided on the topic of firearms making process.

**Reply by ZChris13** [reply]
The most relevant process for the context of gunsmithing in Cataclysm is ECM, which is simply represented by the electrolysis kit in game. IvanPrintsGuns on YouTube has a very good tutorial on this. He only does pistol caliber, for what it's worth.

**Reply by ghost** [reply]
I should note there's significant gameplay arguments against craftable firearms even if get past the crafting hurdles, there's also a situation where they'd be so far restricted to endgame the recipes would go unused, but with my craftable assault rifles and rocket launchers you know where I stand on the issue.

**Reply by blot-runner** [reply]
> I should note there's significant gameplay arguments against craftable firearms even if get past the crafting hurdles, there's also a situation where they'd be so far restricted to endgame the recipes would go unused, but with my craftable assault rifles and rocket launchers you know where I stand on the issue.

For what it's worth, I visualize the whole concept of crafting the non-makeshift firearms as more of a "I LOVE this particular gun, but the RNG seems to hate me when it comes to drops/spawns" thing than a real way of getting an advantage. Given the, lets say _less than optimal_ circumstances, the skill level required to make one (IMO Fabrication 8-9, with a healthy dose of Cooking, Markmanship and Mechanics to accompany it) translates into the stage of the game when you probably already have a hefty gun collection.
That said, I'd love to study the discussion on the topic. Could you maybe direct me to where it was held?

**Reply by ghost** [reply]
The discussion I had was with someone over a year ago in DM's who is now banned and disavowed from CDDA as well as CBN spaces. I'd advise sourcing primary sources on the topic, if all else we can simply do chemical treatment with acid and lead/copper, plus of course any vegetable oil could be a cutting fluid for abstraction's sake.

**Reply by blot-runner** [reply]
> The discussion I had was with someone over a year ago in DM's who is now banned and disavowed from CDDA as well as CBN spaces. I'd advise sourcing primary sources on the topic, if all else we can simply do chemical treatment with acid and lead/copper, plus of course any vegetable oil could be a cutting fluid for abstraction's sake.

Sorry, I meant the discussion about gameplay reasons for/against craftable firearms, not the one about how they're made.

**Reply by ghost** [reply]
Talk to Kheir on the Discord about that specifically.

**Reply by ZChris13** [reply]
> I should note there's significant gameplay arguments against craftable firearms even if get past the crafting hurdles, there's also a situation where they'd be so far restricted to endgame the recipes would go unused, but with my craftable assault rifles and rocket launchers you know where I stand on the issue.

rocket/grenade launchers need to be significantly easier to make

the 3d printed guns need to be more complicated but making them straight upgrades to their competitors would be a good idea imo. should I whip up a pull request for my ideas for this?

**Comment by chaosvolt** [comment]
I've been meaning to remove them for a while but keep forgetting.
